core/parties/helpers_for_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2023 Gobalsky Labs Limited
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package parties_test
+
+import (
+	"testing"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/events"
+	"code.vegaprotocol.io/vega/core/integration/stubs"
+	"code.vegaprotocol.io/vega/core/parties"
+	"code.vegaprotocol.io/vega/core/parties/mocks"
+	"code.vegaprotocol.io/vega/core/snapshot"
+	"code.vegaprotocol.io/vega/core/stats"
+	"code.vegaprotocol.io/vega/core/types"
+	vgrand "code.vegaprotocol.io/vega/libs/rand"
+	"code.vegaprotocol.io/vega/logging"
+	"code.vegaprotocol.io/vega/paths"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testEngine struct {
+	engine *parties.SnapshottedEngine
+	broker *mocks.MockBroker
+}
+
+func assertEqualProfiles(t *testing.T, expected, actual []types.PartyProfile) {
+	t.Helper()
+
+	parties.SortByPartyID(expected)
+	parties.SortByPartyID(actual)
+
+	assert.Equal(t, expected, actual)
+}
+
+func expectPartyProfileUpdatedEvent(t *testing.T, engine *testEngine) {
+	t.Helper()
+
+	engine.broker.EXPECT().Send(gomock.Any()).Do(func(evt events.Event) {
+		_, ok := evt.(*events.PartyProfileUpdated)
+		assert.True(t, ok, "Event should be a PartyProfileUpdated, but is %T", evt)
+	}).Times(1)
+}
+
+func newSnapshotEngine(t *testing.T, vegaPath paths.Paths, now time.Time, engine *parties.SnapshottedEngine) *snapshot.Engine {
```

Comment:
This looks to me like a strange place for this helper... why, in the parties package, are we creating a snapshot engine? This feels to me like it belongs in the snapshot package. As far as unit tests go, we should test the API that the snapshot engine uses/expects with known data, and we can use the same known data in the snapshot engine unit test

Reply:
Only testing the API the snapshot engine is using prevent the test of catching a bunch of error cases. Consuming to API is not enough to prevent certain errors.

Reply:
hence the second part of the test: using the values returned in this test in a unit test that sits in the snapshot package. If we call `GetState()` here, and use those return values in the snapshot test, what are we missing?